core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1651,8 +1639,15 @@ class ReplicaManager(val config: KafkaConfig,
     }
 
     val remoteFetchMaxWaitMs = config.remoteLogManagerConfig.remoteFetchMaxWaitMs().toLong
-    val remoteFetch = new DelayedRemoteFetch(remoteFetchTasks, remoteFetchResults, remoteFetchInfos, remoteFetchMaxWaitMs,
-      fetchPartitionStatus, params, logReadResults, this, responseCallback)
+    val remoteFetch = new DelayedRemoteFetch(remoteFetchTasks,
+                                             remoteFetchResults,
+                                             remoteFetchInfos,
+                                             remoteFetchMaxWaitMs,
+                                             fetchPartitionStatus.toMap.asJava,
```

Comment:
We could initialize `fetchPartitionStatus` as a Map type to reduce unnecessary collection conversions

Reply:
Opened #20769 to address this comment. PTAL.

Reply:
Didn't saw #20768 was also opened. We can merge either one.